File: src/model/pieces/bishop.py
from src.model.pieces.piece import Piece


class Bishop(Piece):
    def __init__(self, color, em):
        key_letter = "b"
        super().__init__(color, key_letter, em)
        # a move is composed of a mov x and mov y and
        # a bool if the movement is linear
        self.moves = [
            ([-1, 1], True),
            ([1, 1], True),
            ([1, -1], True),
            ([-1, -1], True)
        ]

File: src/pieces/bishop.js
import { DirectionalMovement } from '../movement/directionalMovement.js'
import Piece from '../piece.js'

export const bishopMovements = [
  new DirectionalMovement(1, 1),
  new DirectionalMovement(-1, 1),
  new DirectionalMovement(1, -1),
  new DirectionalMovement(-1, -1),
]

export default class Bishop extends Piece {
  constructor(color) {
    super(color, bishopMovements)
  }

  toString() {
    return this.color === 'black' ? '♗' : '♝'
  }
}
